src/main/java/org/logstash/dissect/fields/FieldFactory.java

```diff
@@ -0,0 +1,37 @@
+package org.logstash.dissect.fields;
+
+public final class FieldFactory {
+    private static final String MIXED_PREFIXES = "Field cannot prefix with both Append and Indirect Prefix (%s): %s";
+    private static final String PREFIXED_EMPTY = "Field cannot be a prefix on its own without further text";
+
+    public static Field create(String field) {
+        if (field.isEmpty() || field.startsWith("?")) {
```

Comment:
I feel that, for this release the prefixes and suffixes are few - if/when that changes, this create method will become unwieldy.  One cannot really follow the grammar without reading the docs on the ruby side - I should copy the text of allowable field types to this class.